hyperspy/io_plugins/fei.py

```diff
@@ -316,6 +316,97 @@ def get_xml_info_from_emi(emi_file):
     return objects[:-1]
 
 
+def get_calibration_from_position(position):
+    offset = position[0]
+    for i, x in enumerate(position):
+        if x != position[0]:
+            break
+    if i == len(position) - 1:
+        # No scanning over this axis
+        scale = 0
+        size = 0
+    else:
+        scale = x - position[0]
+        if i == 1:  # Rastering over this dimension first
+            for j, x in enumerate(position[1:]):
+                if x == position[0]:
+                    break
+            size = j + 1
+        else:  # Second rastering dimension
+            size = len(position) / i
+    return {"size": size, "scale": scale, "offset": offset}
+
+
+def get_axes_from_position(header, data):
+    array_shape = []
+    axes = []
+    array_size = int(header["ValidNumberElements"])
+    if data["TagTypeID"][0] == 16706:
```

Comment:
Magic number are bad. Assign it to a constant, or at the very least add a comment.